sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java

```diff
@@ -70,13 +71,18 @@ public RecordNetworkCallPolicy(RecordedData recordedData) {
      * @param redactors The custom redactor functions to apply to redact sensitive information from recorded data.
      */
     public RecordNetworkCallPolicy(RecordedData recordedData, List<Function<String, String>> redactors) {
-        Objects.requireNonNull(recordedData, "'recordedData' cannot be null.");
         this.recordedData = recordedData;
         redactor = new RecordingRedactor(redactors);
+
     }
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // Test is running in LIVE mode so it won't be able to record the network call, skip recording code.
+        if (TestBase.getTestMode() == TestMode.LIVE) {
```

Comment:
Don't we want to record the new data when running in LIVE mode?

Reply:
We record when in `TestMode.RECORD` 😃